jkube-kit/generator/webapp/src/test/java/org/eclipse/jkube/generator/webapp/WebAppGeneratorTest.java

```diff
@@ -80,28 +81,28 @@ public void isApplicableHasGradleWarPluginShouldReturnTrue() {
     assertThat(result).isTrue();
   }
 
-  @Test(expected = IllegalArgumentException.class)
-  public void customizeDoesNotSupportS2iBuildShouldThrowException() {
+  @Test
+  void customizeDoesNotSupportS2iBuildShouldThrowException() {
     // Given
     final Properties projectProperties = new Properties();
     projectProperties.put("jkube.generator.from", "image-to-trigger-custom-app-server-handler");
     when(generatorContext.getRuntimeMode()).thenReturn(RuntimeMode.OPENSHIFT);
     when(generatorContext.getStrategy()).thenReturn(JKubeBuildStrategy.s2i);
     when(generatorContext.getProject().getProperties()).thenReturn(projectProperties);
 
-    // When
-    new WebAppGenerator(generatorContext).customize(Collections.emptyList(), false);
-    // Then - Exception thrown
-    fail();
+    // When & Then
+    assertThatIllegalArgumentException()
+            .isThrownBy(() -> new WebAppGenerator(generatorContext).customize(Collections.emptyList(), false))
+            .withMessageStartingWith("S2I not yet supported for the webapp-generator");
   }
 
   @Test
-  public void customizeWithDefaultHandlerShouldAddImageConfiguration() throws IOException {
+  void customizeWithDefaultHandlerShouldAddImageConfiguration() throws IOException {
     // Given
     final List<ImageConfiguration> originalImageConfigurations = new ArrayList<>();
-    final File buildDirectory = temporaryFolder.newFolder("build");
+    final File buildDirectory = Files.createDirectory(temporaryFolder.resolve("build")).toFile();
     final File artifactFile = new File(buildDirectory, "artifact.war");
-    assertTrue(artifactFile.createNewFile());
+    assertThat(artifactFile.createNewFile()).isTrue();
```

Comment:
Please use nio Files. I'm not sure if you're having trouble with this, or I'm not explaining correctly why we should do so.

Reply:
Sorry I missed this 😔
Let me be clear should I do something like this?
```java
Files.createFile(buildDirectory.toPath().resolve("artifact.war"));
```